server/src/optima/project.py

```diff
@@ -615,7 +615,15 @@ def uploadExcel():
         D.G.workbookname = D.G.projectname + '.xlsx'
         D.G.inputprograms = deepcopy(project.programs)
         D.G.inputpopulations = deepcopy(project.populations)
-        D = updatedata(D, input_programs = project.programs, savetofile = False)
+
+        # Is this the first time? if so then we have to run simulations
+        try:
+            D.S # the cool kids don't reRun when they already have simulations
+            shouldReRun = False
+        except AttributeError:
+            shouldReRun = True
```

Comment:
Yes that's why I've named the variable to expose intention and also commented it.

In fact @Ashalynd suggested a nice one liner similar to that and was not giving the right answer (would not pass the spec).

On the other hand, that "obscure" thing seem to be the preferred Python way as described here:

[EAFP](https://docs.python.org/2/glossary.html#term-eafp)
```
Easier to ask for forgiveness than permission. This common Python coding style assumes the existence of valid keys or attributes and catches exceptions if the assumption proves false. This clean and fast style is characterized by the presence of many try and except statements. The technique contrasts with the LBYL style common to many other languages such as C.
```
Which I understand because

[LBYL](https://docs.python.org/2/glossary.html#term-lbyl)
````
Look before you leap. This coding style explicitly tests for pre-conditions before making calls or lookups. This style contrasts with the EAFP approach and is characterized by the presence of many if statements.

In a multi-threaded environment, the LBYL approach can risk introducing a race condition between “the looking” and “the leaping”. For example, the code, if key in mapping: return mapping[key] can fail if another thread removes key from mapping after the test, but before the lookup. This issue can be solved with locks or by using the EAFP approach.
```